website/src/components/Group/GroupForm.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState, type FC, type FormEvent } from 'react';
+
+import {
+    AddressLineOneInput,
+    AddressLineTwoInput,
+    CityInput,
+    CountryInput,
+    EmailContactInput,
+    GroupNameInput,
+    InstitutionNameInput,
+    PostalCodeInput,
+    StateInput,
+} from './Inputs';
+import useClientFlag from '../../hooks/isClient';
+import type { NewGroup } from '../../types/backend';
+import { ErrorFeedback } from '../ErrorFeedback.tsx';
+
+interface GroupFormProps {
+    title: string;
+    buttonText: string;
+    defaultGroupData?: NewGroup;
+    onSubmit: (group: NewGroup) => Promise<GroupSubmitResult>;
+}
+
+export type GroupSubmitSuccess = {
+    succeeded: true;
+    nextPageHref: string;
+};
+export type GroupSubmitError = {
+    succeeded: false;
+    errorMessage: string;
+};
+export type GroupSubmitResult = GroupSubmitSuccess | GroupSubmitError;
+
+const chooseCountry = 'Choose a country...';
+
+export const GroupForm: FC<GroupFormProps> = ({ title, buttonText, defaultGroupData, onSubmit }) => {
+    const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+
+    const internalOnSubmit = async (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        const formData = new FormData(e.currentTarget);
+
+        const groupName = formData.get(fieldMapping.groupName.id) as string;
+        const institution = formData.get(fieldMapping.institution.id) as string;
+        const contactEmail = formData.get(fieldMapping.contactEmail.id) as string;
+        const country = formData.get(fieldMapping.country.id) as string;
+        const line1 = formData.get(fieldMapping.line1.id) as string;
+        const line2 = formData.get(fieldMapping.line2.id) as string;
+        const city = formData.get(fieldMapping.city.id) as string;
+        const state = formData.get(fieldMapping.state.id) as string;
+        const postalCode = formData.get(fieldMapping.postalCode.id) as string;
+
+        if (country === chooseCountry) {
+            setErrorMessage('Please choose a country');
+            return false;
+        }
+
+        const group: NewGroup = {
+            groupName,
+            institution,
+            contactEmail,
+            address: { line1, line2, city, postalCode, state, country },
+        };
+
+        const result = await onSubmit(group);
+
+        if (result.succeeded) {
+            window.location.href = result.nextPageHref;
+        } else {
+            setErrorMessage(result.errorMessage);
+        }
+    };
+
+    const isClient = useClientFlag();
+
+    return (
+        <div className='p-4 max-w-6xl mx-auto'>
+            <h2 className='title'>{title}</h2>
+
+            {errorMessage !== undefined && (
+                <ErrorFeedback message={errorMessage} onClose={() => setErrorMessage(undefined)} />
```

Comment:
Same thing here, we should be able to do this automatically instead of verbosely

Reply:
You mean line 45 to 53 right?

Reply:
ah yes no clue why it's wrongly annotated, my bad